packages/fxa-auth-server/lib/db/index.ts

```diff
@@ -0,0 +1,1126 @@
+          return sessionTokens;
+        }
+
+        // Prune session tokens
+        try {
+          await this.pruneSessionTokens(uid, expiredSessionTokens);
+        } catch (unusedErr) {
+          // Ignore errors
+        }
+        return sessionTokens;
+      };
+      const promises = [getMysqlSessionTokens()];
+
+      if (this.redis) {
+        promises.push(this.redis.getSessionTokens(uid));
+      }
+
+      const [mysqlSessionTokens, redisSessionTokens = {}] = await Promise.all(
+        promises
+      );
+
+      // for each db session token, if there is a matching redis token
+      // overwrite the properties of the db token with the redis token values
+      const lastAccessTimeEnabled =
+        features.isLastAccessTimeEnabledForUser(uid);
+
+      const sessions = mergeCachedSessionTokens(
+        mysqlSessionTokens,
+        redisSessionTokens,
+        lastAccessTimeEnabled
+      );
+      log.debug('db.sessions.count', {
+        lastAccessTimeEnabled,
+        mysql: mysqlSessionTokens.length,
+        redis: Object.keys(redisSessionTokens).length,
+      });
+      return sessions;
+    }
+
+    async keyFetchToken(id: string) {
+      log.trace('DB.keyFetchToken', { id });
+      const data = await RawKeyFetchToken.findByTokenId(id);
+      if (!data) {
+        throw error.invalidToken('The authentication token could not be found');
+      }
+      return KeyFetchToken.fromId(id, data);
+    }
+
+    async keyFetchTokenWithVerificationStatus(id: string) {
+      log.trace('DB.keyFetchTokenWithVerificationStatus', { id });
+      const data = await RawKeyFetchToken.findByTokenId(id, true);
+      if (!data) {
+        throw error.invalidToken('The authentication token could not be found');
+      }
+      return KeyFetchToken.fromId(id, data);
+    }
+
+    async accountResetToken(id: string) {
+      log.trace('DB.accountResetToken', { id });
+      const data = await RawAccountResetToken.findByTokenId(id);
+      if (!data) {
+        throw error.invalidToken('The authentication token could not be found');
+      }
+      return AccountResetToken.fromHex(data.tokenData, data);
+    }
+
+    async passwordForgotToken(id: string) {
+      log.trace('DB.passwordForgotToken', { id });
+      const data = await RawPasswordForgotToken.findByTokenId(id);
+      if (!data) {
+        throw error.invalidToken('The authentication token could not be found');
+      }
+      return PasswordForgotToken.fromHex(data.tokenData, data);
+    }
+
+    async passwordChangeToken(id: string) {
+      log.trace('DB.passwordChangeToken', { id });
+      const data = await RawPasswordChangeToken.findByTokenId(id);
+      if (!data) {
+        throw error.invalidToken('The authentication token could not be found');
+      }
+      return PasswordChangeToken.fromHex(data.tokenData, data);
+    }
+
+    async accountRecord(email: string, options: { linkedAccounts?: boolean }) {
+      log.trace('DB.accountRecord', { email });
```

Comment:
```suggestion
    async accountRecord(email: string, options?: { linkedAccounts?: boolean }) {
```

options are optional. I'll push a change for this.